map.go

```diff
@@ -497,7 +497,26 @@ func (m *Map) Info() (*MapInfo, error) {
 // Returns an error if the key doesn't exist, see ErrKeyNotExist.
 func (m *Map) Lookup(key, valueOut interface{}) error {
 	valuePtr, valueBytes := makeBuffer(valueOut, m.fullValueSize)
-	if err := m.lookup(key, valuePtr); err != nil {
+	if err := m.lookup(key, valuePtr, 0); err != nil {
+		return err
+	}
+
+	return m.unmarshalValue(valueOut, valueBytes)
+}
+
+// LookupWithFlags retrieves a value from a Map with flags.
+//
+// Passing FLock flag will look up the value of a spin-locked
+// map without returning the lock. This must be specified if the
+// elements contain a spinlock.
+//
+// Calls Close() on valueOut if it is of type **Map or **Program,
+// and *valueOut is not nil.
+//
+// Returns an error if the key doesn't exist, see ErrKeyNotExist.
+func (m *Map) LookupWithFlags(key, valueOut interface{}, flags MapUpdateFlags) error {
```

Comment:
```suggestion
func (m *Map) LookupWithFlags(key, valueOut interface{}, flags MapLookupFlags) error {
```

etc. for the other function.

Reply:
Addressed in 623ff45f69ae33eecd66282db41702eafba6bf66